products/sle/main.pm

```diff
@@ -199,6 +199,39 @@ if (get_var('DEV_IMAGE')) {
     }
 }
 
+#This setting is used to enable all extensions/modules for upgrading tests, generate corresponding extensions/modules list for all addons, all extensions, all modules cases.
+if (check_var('SCC_REGISTER', 'installation') && get_var('ALL_ADDONS')) {
+    my $version = get_required_var("HDDVERSION");
+    #Below $common* store the extensions/modules list in common.
+    my $common_addons_12SP3     = 'sdk,pcm,tcm,wsm';
+    my $common_modules_12SP3    = 'pcm,tcm,wsm';
+    my $common_extensions_12SP3 = 'sdk';
+    my $common_addons_11SP4     = 'ha,geo,sdk';
+    #Below $external* store external extensions/modules list on each ARCH.
+    my %external_addons_12SP3
+      = ('x86_64', 'ha,geo,we,live,asmm,contm,lgm,hpcm', 'ppc64le', 'ha,live,asmm,contm,lgm', 's390x', 'ha,geo,asmm,contm,lgm', 'aarch64', 'hpcm');
+    my %external_modules_12SP3    = ('x86_64', 'asmm,contm,lgm,hpcm', 'ppc64le', 'asmm,contm,lgm', 's390x', 'asmm,contm,lgm', 'aarch64', 'hpcm');
+    my %external_extensions_12SP3 = ('x86_64', 'ha,geo,we,live',      'ppc64le', 'ha,live',        's390x', 'ha,geo',         'aarch64', '');
+    # ALL_ADDONS=1:all addons 2:all extensions 3:all modules
+    if ($version =~ "12-SP3") {
+        set_var("SCC_ADDONS", "$common_addons_12SP3,$external_addons_12SP3{get_var('ARCH')}") if (check_var('ALL_ADDONS', '1'));
+        if (check_var('ALL_ADDONS', '2')) {
+            if ($external_extensions_12SP3{get_var('ARCH')}) {
+                set_var("SCC_ADDONS", "$common_extensions_12SP3,$external_extensions_12SP3{get_var('ARCH')}");
+            }
+            else {
+                set_var("SCC_ADDONS", $common_extensions_12SP3);
+            }
+        }
+        set_var("SCC_ADDONS", "$common_modules_12SP3,$external_modules_12SP3{get_var('ARCH')}") if (check_var('ALL_ADDONS', '3'));
+    }
+    else {
```

Comment:
This can be combined to elsif, as we don't have anything else there. Potentially, in else we can put warning, that no addons defined for this version, so it will be easier to debug.